base/io.jl

```diff
@@ -658,11 +658,11 @@ Commit all currently buffered writes to the given stream.
 flush(io::IO) = nothing
 
 """
-    skipchars(io::IO, predicate; linecomment=nothing)
+    skipchars(io::IO, predicate; linecomment::Char)
 
-Skip forward in `io` until `predicate` returns `false`. If `linecomment`
-is defined, all characters after the `linecomment` character are ignored
-until the next line.
+Advance the stream `io` such that the next-read character will be the first remaining for
+which `predicate` returns `false`. If the keyword argument `linecomment` is specified, all
+characters from that character until the start of the next line are ignored.
 
 ```jldoctext
```

Comment:
Good catch! Poor reviewing of  #22234
Re second comment: That was the purpose of #22234 so the example is correct since it works on master.